src/formatDate.js

```diff
@@ -8,7 +8,66 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldYear = yearFormatAndPlace(fromFormat);
+  const newYear = yearFormatAndPlace(toFormat);
+
+  const formatedDate = [];
+  let year = '';
+
+  const dateParts = date.split(fromFormat[3]);
+  const newDateSeparator = toFormat[3];
+
+  const thresholdYear = 30;
+
+  if (oldYear.format !== newYear.format) {
+    if (oldYear.format.length < newYear.format.length) {
+      year =
+        dateParts[oldYear.place] < thresholdYear
+          ? `20${dateParts[oldYear.place]}`
+          : `19${dateParts[oldYear.place]}`;
+    } else {
+      year = dateParts[oldYear.place].slice(2);
+    }
+  } else {
+    year = dateParts[oldYear.place];
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
```

Comment:
The loop iterates through `toFormat.length - 1`, which is based on the assumption that the last element of `toFormat` is the separator. There should be a validation to ensure that the last element is indeed a separator to prevent potential errors.